app/controllers/sessions_controller.rb

```diff
@@ -11,14 +11,30 @@ def create
       # Save the user id inside the browser cookie. This is how we keep the user
       # logged in when they navigate around our website.
       session[:user_id] = user.id
-      redirect_to dashboard_path
+
+      if request.xhr?
+        render json: { message: "Login Successful", status: 204 }
+      else
+        redirect_to root_path
+      end
+
     else
       assign_login_flash_message
       flash[:danger] = [@notice]
-      redirect_to login_path
+
+      if request.xhr?
+        render :json => { :attachmentPartial => render_to_string('partials/_flash', :layout => false), status: 422 }
```

Comment:
I think the flash message then needs to be cleared or an immediate refresh of the home page shows the error message.